src/main/java/no/entur/uttu/export/netex/producer/common/OrganisationProducer.java

```diff
@@ -132,35 +129,42 @@ private Operator mapOperator(String operatorRef, NetexExportContext context) {
       return new Operator();
     }
 
-    GeneralOrganisation organisation = orgRegOperator.get();
+    Operator organisation = orgRegOperator.get();
 
-    return populateNetexOrganisation(new Operator(), organisation)
+    return organisation
       .withId(getOperatorNetexId(organisation))
       .withCustomerServiceContactDetails(organisation.getContactDetails());
   }
 
-  private <N extends Organisation_VersionStructure> N populateNetexOrganisation(
-    N netexOrg,
-    GeneralOrganisation orgRegOrg
-  ) {
-    netexOrg
-      .withVersion(orgRegOrg.getVersion())
-      .withName(orgRegOrg.getName())
-      .withCompanyNumber(orgRegOrg.getCompanyNumber())
-      .withContactDetails(orgRegOrg.getContactDetails())
-      .withLegalName(orgRegOrg.getLegalName());
-    return netexOrg;
-  }
-
-  private String getOperatorNetexId(GeneralOrganisation organisation) {
+  private String getOperatorNetexId(Operator organisation) {
     return getNetexId(organisation, "Operator");
   }
 
-  private String getAuthorityNetexId(GeneralOrganisation organisation) {
+  private String getAuthorityNetexId(Authority organisation) {
     return getNetexId(organisation, "Authority");
   }
 
-  protected static String getNetexId(GeneralOrganisation organisation, String type) {
+  private <T extends Organisation_VersionStructure> String getNetexId(
+    T organisation,
+    String type
+  ) {
+    if (
+      organisationsOverrides.containsKey(organisation.getId()) &&
+      organisationsOverrides.get(organisation.getId()).containsKey(type)
+    ) {
+      return organisationsOverrides.get(organisation.getId()).get(type);
+    }
+
+    String legacyId = extractLegacyId(organisation, type);
+    if (legacyId != null) return legacyId;
+
+    return organisation.getId();
+  }
+
+  protected static <T extends Organisation_VersionStructure> String extractLegacyId(
```

Comment:
maybe improve this

Reply:
[b85fb01](https://github.com/entur/uttu/pull/432/commits/b85fb01417dbf1cfba7a02732e157f5cd6c61306)